polkadot/node/core/backing/src/lib.rs

```diff
@@ -1105,20 +1232,31 @@ async fn construct_per_relay_parent_state<Context>(
 
 	let session_index = try_runtime_api!(session_index);
 
-	let inject_core_index = request_node_features(parent, session_index, ctx.sender())
+	let validators_arc = per_session_cache
+		.get_or_fetch_validators(session_index, parent, ctx.sender())
+		.await?
+		.unwrap();
```

Comment:
we can't do `unwrap`s because this would bring down the whole subsystem. we need to gracefully handle errors.

Reply:
Correct, I removed `unwrap`s.